[zacharyxzz] iP

zacharyxzz opened the pull request:
# Bob

> “We only live so long, we'll just skip that proof.” – Gilbert Strang

This is my Bob! It's:
- **Text-based**
- *Easy* to learn
- ~~Complicated~~ **Simple** to use

All you need to do is:
1. Download it from [here](https://github.com/nus-cs2103-AY2425S1/ip).
2. Double-click it.
3. Start adding tasks.

Let it manage your tasks for you! 😉 And it's **FREE**!

### Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you're a Java programmer, here’s the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zacharyxzz committed:
level0

zacharyxzz committed:
level1

zacharyxzz committed:
level2

zacharyxzz committed:
level3

zacharyxzz committed:
level4

zacharyxzz committed:
uitest

zacharyxzz committed:
level5

zacharyxzz committed:
level6

zacharyxzz committed:
Implement save feature and error handling.

zacharyxzz committed:
Merge branch 'Level-7'

zacharyxzz committed:
Implement dates and times for Level 8.

zacharyxzz committed:
Implement refactoring for OOP.

zacharyxzz committed:
Refactor into Bob package.

zacharyxzz committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

zacharyxzz committed:
Add JUnit Tests.

zacharyxzz committed:
Package the App as a JAR File.

zacharyxzz committed:
Add JavaDoc comments.

zacharyxzz committed:
Implement find by keyboard feature.

FriedCabbageSalad commented on src/main/java/bob/Parser.java:
```diff
@@ -0,0 +1,130 @@
+            throw new BobException("The description of an event must include '/from' followed by start time and '/to' followed by end time.");
+        }
+        String[] parts = inputSplit[1].split(" /from ", 2);
+        String[] fromTo = parts[1].split(" /to ", 2);
```
Should "fromTo" be renamed to avoid ambiguity?

FriedCabbageSalad commented on src/main/java/bob/Storage.java:
```diff
@@ -0,0 +1,84 @@
+            }
+
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String line;
```
Should "line" be renamed to avoid ambiguity?

FriedCabbageSalad reviewed (commented):
LGTM for Coding Quality, good job! only some minor nits.

yihuididi commented on src/main/java/bob/Parser.java:
```diff
@@ -0,0 +1,130 @@
+        String[] inputSplit = input.split(" ", 2);
+
+        try {
+            switch (inputSplit[0]) {
+                case "bye":
+                    ui.showGoodbye();
+                    System.exit(0);
+                    break;
+                case "list":
+                    ui.showTaskList(tasks);
+                    break;
+                case "mark":
+                    handleMarkCommand(inputSplit, tasks, ui, storage);
+                    break;
+                case "unmark":
+                    handleUnmarkCommand(inputSplit, tasks, ui, storage);
+                    break;
+                case "todo":
+                    handleTodoCommand(inputSplit, tasks, ui, storage);
+                    break;
+                case "deadline":
+                    handleDeadlineCommand(inputSplit, tasks, ui, storage);
+                    break;
+                case "event":
+                    handleEventCommand(inputSplit, tasks, ui, storage);
+                    break;
+                case "delete":
+                    handleDeleteCommand(inputSplit, tasks, ui, storage);
+                    break;
+                case "find":
+                    handleFindCommand(inputSplit, tasks, ui);
+                    break;
+                default:
+                    throw new BobException("I'm sorry, but I don't know what that means :-(");
```
I don't think this follows the coding standard for CS2103 switch cases.

yihuididi commented on src/main/java/bob/Parser.java:
```diff
@@ -0,0 +1,130 @@
+        storage.save(tasks);
+    }
+
+    private static void handleUnmarkCommand(String[] inputSplit, TaskList tasks, Ui ui, Storage storage) throws BobException {
```
Might want to split the line because line extends beyond 120 characters.

yihuididi commented on src/main/java/bob/Storage.java:
```diff
@@ -0,0 +1,84 @@
+package bob;
+import java.io.*;
```
I think importing wildcards is generally not recommended.

yihuididi commented on src/test/java/bob/TaskListTest.java:
```diff
@@ -0,0 +1,55 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
I think static imports should be placed above normal imports.

yihuididi reviewed (commented):
Overall good job! The code is well organised. Just a few nit picking on Coding Standards violations.

zacharyxzz committed:
Add error handling for deadline and event input

zacharyxzz committed:
Add GUI for Level 10

zacharyxzz committed:
Integrate FXML partially

zacharyxzz committed:
Implement GUI with FXML

zacharyxzz committed:
Add assertions

Assertions were added to several methods to ensure validity of parameters.

zacharyxzz committed:
Improve code quality

Fixed any outstanding issues CodeQuality issues that were missed from previous comments.

zacharyxzz committed:
Tag commit

zacharyxzz committed:
Merge pull request #3 from zacharyxzz/Assertions

Assertions

zacharyxzz committed:
Merge pull request #4 from zacharyxzz/branch-A-CodeQuality

branch-A-CodeQuality

zacharyxzz committed:
Add extension feature to track expenses

Users can now track expenses with the expense command, which takes in a name and an amount.

zacharyxzz committed:
Update tag to meet grader requirement

zacharyxzz committed:
Add Ui image

zacharyxzz committed:
Update user guide

zacharyxzz committed:
Fix event and deadline feature

zacharyxzz committed:
Re-added submodule javafx-tutorial

zacharyxzz committed:
Re-add submodule javafx-tutorial

zacharyxzz committed:
Merge branch 'master' of https://github.com/zacharyxzz/ip

zacharyxzz committed:
Ammend user guide

zacharyxzz committed:
Reformat long lines

zacharyxzz committed:
Fix taks class visibility issue

zacharyxzz committed:
Fix formatting for main class

zacharyxzz committed:
Reformat parser class long lines

zacharyxzz committed:
Reformat long lines in storage class

zacharyxzz committed:
Remove submodule javafx-tutorial
